retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -129,80 +135,55 @@ public Request request() {
             return request;
         }
 
-        private synchronized CompletableFuture<AggregatedHttpMessage> createRequest() {
-            if (executed) {
-                throw new IllegalStateException("Already Executed");
+        private synchronized void createRequest() {
+            if (httpResponse != null) {
+                throw new IllegalStateException("executed already");
             }
-            final CompletableFuture<AggregatedHttpMessage> future = doCall(callFactory.httpClient, request)
-                    .aggregate();
-            executed = true;
-            return future;
+            executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING);
```

Comment:
Perhaps we could move this `compareAndSet()` to the beginning of `createRequest()` and remove the `synchronized` modifier? e.g.

```java
private void createRequest() {
    if (!executionStateUpdater.compareAndSet(this, IDLE, RUNNING)) {
        throw new IllegalStateException(...);
    }
    ...
}
```